### PR TITLE
579 new ecosystems dictionary

### PR DESCRIPTION
**Pull request by rpoletaev**

### CODE REVIEW

File: packages/api/api.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ type apiData struct {
 	result      interface{}
 	params      map[string]interface{}
 	ecosystemId int64
+	systemName  string
```
Comment (ghost) [inline packages/api/api.go]: maybe ecosystemName?

File: packages/api/content.go [hunk 2]
```diff
@@ -53,6 +53,7 @@ func initVars(r *http.Request, data *apiData) *map[string]string {
 	vars[`ecosystem_id`] = converter.Int64ToStr(data.ecosystemId)
 	vars[`key_id`] = converter.Int64ToStr(data.keyId)
 	vars[`isMobile`] = data.isMobile
+	vars[`system_name`] = data.systemName
```
Comment (ghost) [inline packages/api/content.go]: ecosystem_name

File: packages/api/login.go [hunk 3]
```diff
@@ -122,27 +134,43 @@ func login(w http.ResponseWriter, r *http.Request, data *apiData, logger *log.En
 		founder = converter.StrToInt64(sp.Value)
 	}
 
-	result := loginResult{EcosystemID: converter.Int64ToStr(state), KeyID: converter.Int64ToStr(wallet),
-		Address: address, IsOwner: founder == wallet, IsNode: conf.Config.KeyID == wallet,
-		IsVDE: model.IsTable(fmt.Sprintf(`%d_vde_tables`, state))}
+	result := loginResult{
+		EcosystemID: converter.Int64ToStr(state),
```
Comment (ghost) [inline packages/api/login.go]: if "state" is local variable - replace it with "ecosystem"

File: packages/api/login.go [hunk 4]
```diff
@@ -122,27 +134,43 @@ func login(w http.ResponseWriter, r *http.Request, data *apiData, logger *log.En
...
 		expire = jwtExpire
 	}
+
 	var isMobile string
 	if data.params[`mobile`].(string) == `1` || data.params[`mobile`].(string) == `true` {
 		isMobile = `1`
 	}
+
+	var ecosystem model.Ecosystem
+	if err := ecosystem.Get(state); err != nil {
```
Comment (ghost) [inline packages/api/login.go]: the smae about "state"

File: packages/model/system_states.go [hunk 5]
```diff
@@ -1,29 +1,43 @@
...
+// only first ecosystem has this entity
+func (sys *Ecosystem) TableName() string {
+	return "1_ecosystems"
 }
 
 // GetAllSystemStatesIDs is retrieving all system states ids
 func GetAllSystemStatesIDs() ([]int64, error) {
-	states := new([]SystemState)
+	if !IsTable("1_ecosystems") {
+		return nil, nil
```
Comment (ghost) [inline packages/model/system_states.go]: why error is nil? This table is have to be created from first migration. If it isn't exists if should be logged.

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/api/login.go (hunk 3 above)
Comment (potehinre) [reply packages/api/login.go]: ecosystemID

File: packages/migration/ecosysystem.go [hunk 6]
```diff
@@ -1140,6 +1139,16 @@ If("#key_id#" == EcosysParam("founder_account")){
...
 	SchemaFirstEcosystem = `
+	DROP TABLE IF EXISTS "1_ecosystems";
+	CREATE TABLE "1_ecosystems" (
+			"id" bigint NOT NULL DEFAULT '0',
+			"name"	varchar(255) NOT NULL DEFAULT '',
+			"is_valued" bigint NOT NULL DEFAULT '0'
+	);
+	ALTER TABLE ONLY "1_ecosystems" ADD CONSTRAINT "1_ecosystems_pkey" PRIMARY KEY ("id");
+
+	INSERT INTO "1_ecosystems" ("id", "name", "is_valued") VALUES ('1', 'base ecosystem', 1);
```
Comment (potehinre) [inline packages/migration/ecosysystem.go]: not base, but platform ecosystem

File: packages/migration/ecosysystem.go [hunk 7]
```diff
@@ -2036,5 +2052,24 @@ If("#key_id#" == EcosysParam("founder_account")){
 			DBUpdate("delayed_contracts", $Id, "counter,block_id", counter, block_id)
 			CallContract($cur["contract"], nil)
 		}
-	}','%[1]d', 'ContractConditions("MainCondition")');`
+	}','%[1]d', 'ContractConditions("MainCondition")'),
+	('31', 'contract EditEcosystemName {
+		data {
+			SystemID int
```
Comment (potehinre) [inline packages/migration/ecosysystem.go]: EcosystemID

File: packages/migration/ecosysystem.go [hunk 8]
```diff
@@ -2036,5 +2052,24 @@ If("#key_id#" == EcosysParam("founder_account")){
...
+			NewName string
+		}
+		conditions {
+			var rows array
+			rows = DBFind("1_ecosystems").Where("id = ?", $SystemID)
```
Comment (potehinre) [inline packages/migration/ecosysystem.go]: same

File: packages/migration/ecosysystem.go [hunk 9]
```diff
@@ -2036,5 +2052,24 @@ If("#key_id#" == EcosysParam("founder_account")){
...
+			if !Len(rows) {
+				error Sprintf("Ecosystem %%d does not exist", $SystemID)
```
Comment (potehinre) [inline packages/migration/ecosysystem.go]: same

File: packages/migration/ecosysystem.go [hunk 10]
```diff
@@ -2036,5 +2052,24 @@ If("#key_id#" == EcosysParam("founder_account")){
...
+			}
+		}
+		action {
+			EditEcosysName($SystemID, $NewName)
```
Comment (potehinre) [inline packages/migration/ecosysystem.go]: same

File: packages/model/system_states.go [hunk 11]
```diff
@@ -1,29 +1,43 @@
...
+	}
+
+	states := new([]Ecosystem)
```
Comment (potehinre) [inline packages/model/system_states.go]: ecosystems

File: packages/smart/smart.go [hunk 12]
```diff
@@ -333,11 +333,14 @@ func LoadContracts(transaction *model.DbTransaction) (err error) {
 	var states []map[string]string
 	var prefix []string
 	prefix = []string{`system`}
-	states, err = model.GetAll(`select id from system_states order by id`, -1)
-	if err != nil {
-		log.WithFields(log.Fields{"type": consts.DBError, "error": err}).Error("selecting ids from system_states")
-		return err
+	if model.IsTable("1_ecosystems") {
+		states, err = model.GetAll(`select id from "1_ecosystems" order by id`, -1)
```
Comment (potehinre) [inline packages/smart/smart.go]: move that to model please

**Review by potehinre — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by potehinre — APPROVED** [review]

**Pull request merged**